### ISSUE TITLE
Adapt framework detection to .NET 5

### ISSUE THREAD

**Issue by gregkalapos** [elastic/apm-agent-dotnet#945]
`RuntimeInformation.FrameworkDescription` on .NET 5 reports: `.NET 5.0.0-preview.7.20364.11`

Our `PlatformDetection.cs` isn't prepared for this - we need to adapt it. Currently it reports  "N/A".

Logs when this happens:

```
[2020-09-01 13:42:01.893 +02:00][Error] - {Service} Failed to detect whether the current .NET runtime is .NET Full Framework, Mono or .NET Core - `N/A' will be used as the current .NET runtime name
[2020-09-01 13:42:01.893 +02:00][Error] - {Service} Failed to detect whether the current .NET runtime is .NET Full Framework or .NET Core - `N/A' will be used as the current .NET runtime version
```

**Comment by lKoTl** [comment]
This is a must have!
Encountered that issue :)

**Comment by gregkalapos** [comment]
Fixed in #964.
